File: 0007-reverse-integer.py
class Solution:
    def reverse(self, x: int) -> int:
        flag = 1
        if x < 0:
            flag = -flag
            x = -x
        result = 0
        while x != 0:
            current_value = x % 10
            x = x // 10
            result = result * 10 + current_value

        result *= flag

        if -2**31 <= result <= 2**31 - 1:
            return result

        return 0


if __name__ == "__main__":
    print(Solution().reverse(-123))

File: 0007-reverse-integer.c

#include <stdio.h>
#include <limits.h>
#include <stdint.h>

int reverse(int x){
    long result = 0;
    short temp = 0;

    while (x != 0) {
        if (result < INT_MIN / 10 || result > INT_MAX / 10) {
            return 0;
        }
        temp = x % 10;
        x = x / 10;
        result = result * 10 + temp;
    }
    return result;
}

int main(int argc, char *argv[]) {
    int num = 123;
    printf("%d\n", reverse(num));
    return 0;
}
